### PR TITLE
[LEGION]: Port GLTF Module

### PR DESCRIPTION
**Pull request by grzi**
## Description
This PR port the GLTF Module to Legion.

What has been done : 

- move from format trait to a distill importer
- load the gltf embeded & glb file.
- successfully return a scene prefab & some other assets
- add the scene prefab to the world adds the entities to the world.
- Camera, Transform, Light are working
- create/modify an example to be able to test the importer. (Move to gltf_scene to avoir conflict in cargo run AND simplify the example to focus on what matter here : GLTF import) 
- be able to add an handle as a component to an entity in the scene prefab
- load materials and use them in the meshes
- multi primitives
- handle skinning
- bounding box
- load animations 
- Polishing and debugging

## How Has This Been Tested?
Unit tests, example

## Screenshots (if appropriate):
screen

## Issues 
- Lots of fake impl of trait to be able to add components to the main prefab... What to do with these...

## Checklist:

- [X] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.
- [X] My code follows the code style of this project.
- [ ] If my change required a change to the documentation I have updated the documentation accordingly.
- [ ] I have updated the content of the book if this PR would make the book outdated.
- [X] I have added tests to cover my changes.
- [X] My code is used in an example.

### CODE REVIEW

**Comment by grzi** [comment]
I still need to clean some code && updates I made but I wanted to have this draft to keep track of the CI.

File: amethyst_assets/src/storage.rs [hunk 1]
```diff
@@ -131,6 +131,14 @@ impl<A> AssetStorage<A> {
         self.get_asset_state(handle.load_handle()).map(|a| &a.asset)
     }
 
+    pub fn pop<T: AssetHandle>(&mut self, handle: &T) -> Option<A> {
```
Comment (grzi) [inline amethyst_assets/src/storage.rs:134]: Need to remove that, no longer needed since my last update

**Comment by grzi** [comment]
**DONE:**
- move from format trait to a distill importer
- load the gltf embeded & glb file.
- successfully return a scene prefab & some other assets
- add the scene prefab to the world adds the entities to the world. 
- Camera, Transform, Light are working 
- create/modify an example to be able to test the importer.
- be able to add an handle as a component to an entity in the scene prefab
- load materials and use them in the meshes
- multi primitives

**In progress:**

**TODO:*
- load animations
- handle skinning

**Comment by grzi** [comment]
I am fixing issues with animation, should be ready in a day or two

**Review by AlveLarsson — APPROVED** [review]
Runs well on Linux for me,
one thing is to clean up the 8 warnings generated in amethyst_gltf and some left in the example and animation crates.

otherwise, this checks out, awesome work!

**Comment by ezpuzz** [comment]
This is a fantastic improvement!  I look forward to seeing the community response to this.

**Review by ezpuzz — APPROVED** [review]

**Pull request merged**